Makefile

```
@@ -10,22 +10,25 @@ MANPREFIX = $(PREFIX)/share/man
 AUTORELOAD = inotify
 
 # enable features requiring giflib (-lgif)
-HAVE_GIFLIB = 1
-
+HAVE_LIBGIF ?= 0
 # enable features requiring libexif (-lexif)
-HAVE_LIBEXIF = 1
-
-cflags = -std=c99 -Wall -pedantic $(CFLAGS)
-cppflags = -I. $(CPPFLAGS) -D_XOPEN_SOURCE=700 \
-  -DHAVE_GIFLIB=$(HAVE_GIFLIB) -DHAVE_LIBEXIF=$(HAVE_LIBEXIF) \
+HAVE_LIBEXIF ?= 0
+
+ifndef OPTIONAL_LIBS
+	ifeq ($(HAVE_LIBEXIF), 1)
+		OPTIONAL_LIBS += -lexif
+	endif
+	ifeq ($(HAVE_LIBGIF), 1)
+		OPTIONAL_LIBS += -lgif
+	endif
+endif
+
+CFLAGS ?= -std=c99 -Wall -pedantic
```

These aren't optimizations, so we should have them set by default.

```suggestion
CFLAGS += -std=c99 -Wall -pedantic
```

---

They are set by default either way. The question is if they should just be default or extend whatever the use has set. I generally lead towards giving people the ability to change build flags with modifying the makefile. One reason people may want to turn off -Wall is to silence harmless warning messages like ` warning: function might return no value: 'cg_quit'`. Probably not a big deal but just giving the option cause it is free to do so.

---

> They are set by default either way.

Not if the environment has `CFLAGS` defined.

---

I see the confusion. By default, I meant if CFLAGS isn't set in the env or directly passed into make.

---

Btw users can set `-w` in their CFLAGS to disable warnings.

---

> if CFLAGS isn't set in the env
Yeah, that's what I meant. If it's defined in the env then those flags will get overridden.